File: cses/MexGridConstruction.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<vector<int> > ans(n, vector<int> (n));

    for(int i = 0; i < n; i++) {
        for(int j = 0; j < n; j++) {
            unordered_set<int> seen;
            for(int k = 0; k < i; k++) {
                seen.insert(ans[k][j]);
            }
            for(int k = 0; k < j; k++) {
                seen.insert(ans[i][k]);
            }
            for(int k = 0; k < 2 * n; k++) {
                if(seen.find(k) == seen.end()) {
                    ans[i][j] = k;
                    break;
                }
            }
        }
    }

    for(auto &row : ans) {
        for(int x : row) {
            cout << x << " ";
        }
        cout << "\n";
    }

    return 0;
}

File: cses/MexGridConstruction.py
n = int(input())

ans = [[0] * n for _ in range(n)]

for i in range(n):
    for j in range(n):
        s = set()
        for k in range(i):
            s.add(ans[k][j])

        for k in range(j):
            s.add(ans[i][k])

        for k in range(2 * n):
            if k not in s:
                ans[i][j] = k
                break

for row in ans:
    print(*row)
